smack-integration-test/src/main/java/org/igniterealtime/smack/inttest/util/SimpleResultSyncPoint.java

```diff
@@ -29,4 +31,14 @@ public void signalFailure() {
     public void signalFailure(String failureMessage) {
         signal(new Exception(failureMessage));
     }
+
+    public static void assertSuccess(SimpleResultSyncPoint ResultSyncPoint, long timeout, String message) throws InterruptedException {
```

Comment:
I think it would probably be more consistent to call this method `assertResult()` or `assertHasResult()`. The result must not always indicate success.

Reply:
Thinking of it, this is basically, and therefore should become, `ResultSyncPoint.waitForResult()` with a second parameter `String message`.

Reply:
I think that's the difference between `ResultSyncPoint` and `SimpleResultSyncPoint`. In the 'simple' version, any result _does_ indicate success. Failures are represented by an Exception being thrown.

The `assertResult()` (or possibly `assertResultNotNull()`, `assertResultNull()` and `assertResultEquals()`) would make for nice candidates in the 'non-simple' version of the sync point implementation. Pragmatically, I've hardly seen these be used, so I was trying to prevent adding unneeded code in this PR.

Reply:
> In the 'simple' version, any result _does_ indicate success.

No, `SimpleResultSyncPoint` returns a Boolean result, which very well could indicate an expected failure result.

If you really want to go with an junit-assert-like API, then have a look at https://github.com/igniterealtime/Smack/pull/583

Reply:
Thought: what if we'd change `SimpleResultSyncPoint` to extend from `ResultSyncPoint<Void, Exception>` instead of `ResultSyncPoint<Boolean, Exception>`?

Reply:
I've just pushed a commit that does exactly that. It was deceptively easy, requiring only code changes in places that are, as far as I can tell, unreachable code bits anyway (which in itself is smelly).

Reply:
I've now pulled in the API changes from #583 

Pondering if it'd be good to rename the method to `assertHasResult` or `assertNotNullResult` to allow for later additions like `assertResultEquals` - far from a hill for me to die on though.